layerindex/tools/import_otherdistro.py

```diff
@@ -0,0 +1,789 @@
+        specpath = os.path.relpath(os.path.dirname(specfile), reldir)
+        recipe, created = ClassicRecipe.objects.get_or_create(layerbranch=layerbranch, filepath=specpath, filename=specfn)
+        if created:
+            logger.info('Importing %s' % specfn)
+        elif recipe.deleted:
+            logger.info('Restoring and updating %s' % specpath)
+            recipe.deleted = False
+        else:
+            logger.info('Updating %s' % specpath)
+        recipe.layerbranch = layerbranch
+        recipe.filename = specfn
+        recipe.filepath = specpath
+        update_recipe_file(specfile, recipe, reldir)
+        recipe.save()
+        existingentry = (specpath, specfn)
+        if existingentry in existing:
+            existing.remove(existingentry)
+        if updateobj:
+            rupdate, _ = ComparisonRecipeUpdate.objects.get_or_create(update=updateobj, recipe=recipe)
+            rupdate.meta_updated = True
+            rupdate.save()
+        recipes.append(recipe)
+    return recipes
+
+
+def import_pkgspec(args):
+    utils.setup_django()
+    import settings
+    from layerindex.models import LayerItem, LayerBranch, Recipe, ClassicRecipe, Machine, BBAppend, BBClass, ComparisonRecipeUpdate
+    from django.db import transaction
+
+    ret, layerbranch = check_branch_layer(args)
+    if ret:
+        return ret
+
+    updateobj = get_update_obj(args)
+    logdir = getattr(settings, 'TASK_LOG_DIR')
+    if updateobj and updateobj.task_id and logdir:
+        pwriter = utils.ProgressWriter(logdir, updateobj.task_id, logger=logger)
+    else:
+        pwriter = None
+
+    metapath = args.pkgdir
+
+    try:
+        with transaction.atomic():
+            layerrecipes = ClassicRecipe.objects.filter(layerbranch=layerbranch)
+            existing = list(layerrecipes.filter(deleted=False).values_list('filepath', 'filename'))
+            count = import_specdir(metapath, layerbranch, existing, updateobj, pwriter)
+
+            if count == 0:
+                logger.error('No spec files found in directory %s' % metapath)
+                return 1
+
+            if args.relative_path:
+                layerbranch.local_path = os.path.relpath(metapath, args.relative_path)
+
+            if existing:
+                fpaths = sorted(['%s/%s' % (pth, fn) for pth, fn in existing])
+                logger.info('Marking as deleted:\n  %s' % '\n  '.join(fpaths))
+                for entry in existing:
+                    layerrecipes.filter(filepath=entry[0], filename=entry[1]).update(deleted=True)
+
+            if args.description:
+                logger.debug('Setting description to "%s"' % args.description)
+                branch = layerbranch.branch
+                branch.short_description = args.description
+                branch.save()
+                layer = layerbranch.layer
+                layer.summary = args.description
+                layer.save()
+
+            layerbranch.vcs_last_fetch = datetime.now()
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

The naive datetime objects are treated by many datetime methods as local times, it is preferred to use aware datetimes to represent times in UTC. The recommended way to create an aware datetime object representing a specific timestamp in UTC is by passing `tzinfo` as an argument to the method.

[Learn more](https://docs.python.org/3/library/datetime.html#aware-and-naive-objects)